### ISSUE TITLE
claim: `head`

### ISSUE THREAD

**Issue by shadowninja55** [vlang/coreutils#11]

**Comment by billpcs** [comment]
Are you still interested?

**Comment by billpcs** [comment]
Just something I noticed by playing around with `head`.

The `--lines` and `--bytes` options are mutually exclusive. However `head` accepts both of them with no error if both are passed.

It looks like there is no predefined precedence between the two. What happens is that the _last_ option to be passed will take effect.

So `head --lines=2 --bytes=4` will show the first 4 bytes.

**Comment by JalonSolov** [comment]
Many Unix tools work that way.

@shadowninja55 has been away from V for a while - I don't think he'll mind if you pick this up.

**Comment by shadowninja55** [comment]
sorry for the late response, @billpcs feel free to do this

**Comment by billpcs** [comment]
Thanks @shadowninja55! I have still not decided to commit to doing it for sure, but I _have_ written some code on it.

I would like to ask a design question (if this is not the right place, feel free to point me elsewhere)

Some `coreutils` receive a command line option that accepts an argument, e.g for `nl` you can use `--body-numbering=XXX` to set the numbering style.

When a user for some reason calls `nl --body-numbering=t --body-numbering=a` then the **last** argument takes effect. In this case `nl` will run for `--body-numbering=a`.

However, our built-in `flag` parser will save the **first** argument, so for the above command we would get the `--body-numbering=t` back, as a result of the parsing.

I verified it, that currently the GNU `nl` and the **v/coreutils** `nl` behave differently in that matter.

Is it something I should be concerned with while implementing `head`, or is it a corner case? 

In other words how does it fall into the README proposition:
> We are looking for solid, working implementations of the commands, not 100% 1-to-1 parity, especially not quirks and unintended side-effects.

**Comment by JalonSolov** [comment]
I think it's a corner case that you don't need to worry about, at least for the first version of the code.  It can be always be fixed later if required, and may be something that could be fixed in the `flag` module rather than each of these tools.  That's _if_ we decide it is something that needs to be fixed at all.
